pipeline/inferrer/inference_manager/src/main/scala/uk/ac/wellcome/platform/inference_manager/services/InferenceManagerWorkerService.scala

```diff
@@ -38,48 +50,94 @@ class InferenceManagerWorkerService[Destination](
           case (_, image) => image
         }
         .via(imageDownloader.download)
-        .via(createRequest)
+        .via(createRequests)
         .via(requestPool.asContextFlow)
         .via(unmarshalResponse)
-        .via(augmentInput)
+        .via(collectAndAugment)
         .via(sendAugmented)
         .asSource
         .map { case (_, msg) => msg }
     )
 
-  private def createRequest[Ctx] =
-    FlowWithContext[DownloadedImage, Ctx].map { image =>
-      (inferrerAdapter.createRequest(image), image)
+  private def createRequests[Ctx] =
+    FlowWithContext[DownloadedImage, Ctx].mapConcat { image =>
+      inferrerAdapters.map { inferrerAdapter =>
+        (
+          inferrerAdapter.createRequest(image),
+          (image, inferrerAdapter)
+        )
+      }
     }
 
   private def unmarshalResponse[Ctx] =
-    FlowWithContext[(Try[HttpResponse], DownloadedImage), Ctx]
+    FlowWithContext[
+      (Try[HttpResponse], (DownloadedImage, InferrerAdapter)),
+      Ctx]
       .map {
-        case result @ (_, image) =>
+        case result @ (_, (image, _)) =>
           imageDownloader.delete.runWith(Source.single(image))
           result
       }
       .mapAsync(parallelism) {
-        case (Success(response), image) =>
-          inferrerAdapter
+        case (Success(response), (image, adapter)) =>
+          adapter
             .parseResponse(response)
+            .map(Success(_))
             .recover {
               case e: Exception =>
                 response.discardEntityBytes()
-                throw e
+                Failure(e)
+            }
+            .map { response =>
+              AdapterResponseBundle(
+                image,
+                adapter,
+                response
+              )
             }
-            .map((image, _))
         case (Failure(exception), _) =>
           Future.failed(exception)
       }
 
-  private def augmentInput[Ctx] =
-    FlowWithContext[
-      (DownloadedImage, Option[inferrerAdapter.InferrerResponse]),
-      Ctx]
-      .map {
-        case (image, response) =>
-          inferrerAdapter.augmentInput(image, response)
+  private def collectAndAugment[Ctx <: Message] =
+    FlowWithContext[AdapterResponseBundle[DownloadedImage], Ctx]
+      .via {
+        Flow[(AdapterResponseBundle[DownloadedImage], Ctx)]
+          .groupBy(parallelism, _ match {
+            case (_, msg) => msg.messageId()
+          }, allowClosedSubstreamRecreation = true)
+          .groupedWithin(inferrerAdapters.size, maxInferrerWait)
+          .map { elements =>
+            elements.filter {
+              case (AdapterResponseBundle(_, _, Failure(_)), _) => false
```

Comment:
Don't understand all the requirements, but do we need to retry failures or anything, or is it fine just ignoring them like this?